### PR TITLE
Update styled-component docs

### PR DESCRIPTION
**Pull request by kristojorg**
- Made it more clear in the docs that to extend `Document` you must create a file `./Pages/_document.js`, which will be automatically consumed by next. 

- Added a note to the `styled-components` example about an existing issue when SSR without rendering any styled components. I have submitted a PR to them to fix.

Side note: this is my second PR ever, after the first one to `styled-components` 10 mins ago... please let me know if I'm doing something wrong!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -410,7 +410,7 @@ Supported options:
   <ul><li><a href="./examples/with-styled-components">Styled components custom document</a></li></ul>
 </details></p>
 
-Pages in `Next.js` skip the definition of the surrounding document's markup. For example, you never include `<html>`, `<body>`, etc. But we still make it possible to override that:
+Pages in `Next.js` skip the definition of the surrounding document's markup. For example, you never include `<html>`, `<body>`, etc. To override that default behavior, you must create a file at `./Pages/_document.js`, where you can extend the `Document` class:
```
Comment (rauchg) [inline README.md]: It's not `Pages`, it's `pages`

**Comment by rauchg** [comment]
Great stuff @kristojorg ! Hopefully we can remove that caveat soon :) Only one nitpick before I merge

**Comment by kristojorg** [comment]
Cool, updated that!

**Comment by timneutkens** [comment]
Thank you very much 🙌🏻

**Pull request merged**